Lib/concurrent/futures/_base.py

```diff
@@ -170,6 +170,17 @@ def _create_and_install_waiters(fs, return_when):
 
     return waiter
 
+
+def _yield_future(fs, waiter, ref_collect=()):
+    while fs:
+        with fs[0]._condition:
+            fs[0]._waiters.remove(waiter)
+
+        for future_list in ref_collect:
+            future_list.remove(fs[0])
+        yield fs.pop(0)
```

Comment:
In this case fs is always a list, however I can modify this function to yield last element. I think order here does not matter. If order matter, reversing list is the option. What do you think?